### PR TITLE
Add config for system-contract-pubkey

### LINKED ISSUE #3676: Add config for system-contract-pubkey
## Overview

~~**Bloked by https://github.com/rchain/rchain/pull/3485**~~ Resolved

Currently, in #3485 the owner key of the registry is hard-coded. We need to config this in the config file and command line for genesis like #3571 .

### Design

1. add one more `system-contract-pubkey`(maybe some other names) to https://github.com/rchain/rchain/blob/ced5a7251bcb8c2f9a995c843d82b12d79845ebd/node/src/main/resources/defaults.conf#L250-L301
2. add another same option in https://github.com/rchain/rchain/blob/ced5a7251bcb8c2f9a995c843d82b12d79845ebd/node/src/main/scala/coop/rchain/node/configuration/commandline/Options.scala#L150-L157
3. Create an case class like https://github.com/rchain/rchain/blob/28b303347a36e163d20c646c3d9ffb48cea0208b/casper/src/main/scala/coop/rchain/casper/genesis/contracts/ProofOfStake.scala#L10-L38 but for https://github.com/rchain/rchain/blob/28b303347a36e163d20c646c3d9ffb48cea0208b/casper/src/main/resources/Registry.rho and render the key in https://github.com/rchain/rchain/blob/28b303347a36e163d20c646c3d9ffb48cea0208b/casper/src/main/resources/Registry.rho#L65-L67 to the config value

### PR DESCRIPTION
**Pull request by maximus19910112**
## Overview
Added new command - line parameter according to the issue #3676 

### Notes
1. Added new command line parameter `system-contract-pub-key`
2. Added module `Registry` (analog of `ProofOfStake`) for `system-contract-pub-key`

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: casper/src/main/scala/coop/rchain/casper/genesis/contracts/ConfigRegistry.scala [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+final case class ConfigRegistry(
+    minimumBond: Long,
+    maximumBond: Long,
+    validators: Seq[Validator],
+    epochLength: Int,
+    quarantineLength: Int,
+    numberOfActiveValidators: Int,
+    posMultiSigPublicKeys: List[String],
+    posMultiSigQuorum: Int,
+    posVaultPubKey: String,
```
Comment (zsluedem) [inline casper/src/main/scala/coop/rchain/casper/genesis/contracts/ConfigRegistry.scala]: the above parameters are not needed.

File: casper/src/main/scala/coop/rchain/casper/genesis/contracts/ConfigRegistry.scala [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  def publicKeys(posMultiSigPublicKeys: List[String]): String = {
+    val indentBrackets = 12
+    val indentKeys     = indentBrackets + 2
+    val pubKeyItems = posMultiSigPublicKeys
+      .map(pk => s"""${" " * indentKeys}"$pk".hexToBytes()""")
+      .mkString(",\n")
+    s"""[\n$pubKeyItems\n${" " * indentBrackets}]"""
+  }
```
Comment (zsluedem) [inline casper/src/main/scala/coop/rchain/casper/genesis/contracts/ConfigRegistry.scala]: The above functions are not needed.

File: casper/src/main/scala/coop/rchain/casper/genesis/contracts/ConfigRegistry.scala [hunk 3]
```diff
@@ -0,0 +1,64 @@
+package coop.rchain.casper.genesis.contracts
+
+import coop.rchain.models.NormalizerEnv
+import coop.rchain.rholang.build.CompiledRholangTemplate
+import coop.rchain.shared.Base16
+
+final case class ConfigRegistry(
```
Comment (zsluedem) [inline casper/src/main/scala/coop/rchain/casper/genesis/contracts/ConfigRegistry.scala]: This case class should be used in https://github.com/rchain/rchain/blob/dev/casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala#L54
like `ProofOfStake`.

File: casper/src/main/scala/coop/rchain/casper/genesis/contracts/ConfigRegistry.scala [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    systemContractPubKey: String =
+      "04e2eb6b06058d10b30856043c29076e2d2d7c374d2beedded6ecb8d1df585dfa583bd7949085ac6b0761497b0cfd056eb3d0db97efb3940b14c00fff4e53c85bf"
```
Comment (zsluedem) [inline casper/src/main/scala/coop/rchain/casper/genesis/contracts/ConfigRegistry.scala]: We don't need default value here.

**Review by zsluedem — CHANGES_REQUESTED** [review]
https://github.com/maximus19910112/rchain/blob/b5aec2218cb29af8f7828c0d8dfd1b9393d95846/casper/src/main/scala/coop/rchain/casper/engine/NodeLaunch.scala#L58-L62
you should log some warning like you did previously.

File: casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala [hunk 5]
```diff
@@ -35,7 +36,8 @@ object Genesis {
   def defaultBlessedTerms(
       posParams: ProofOfStake,
       vaults: Seq[Vault],
-      shardId: String
+      shardId: String,
+      configRegistry: ConfigRegistry
```
Comment (nzpr) [inline casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala]: Better place this next to `ProofOfStake` as they are both compiled Rholang templates

File: casper/src/main/scala/coop/rchain/casper/genesis/contracts/ConfigRegistry.scala [hunk 6]
```diff
@@ -0,0 +1,12 @@
+package coop.rchain.casper.genesis.contracts
+
+import coop.rchain.models.NormalizerEnv
+import coop.rchain.rholang.build.CompiledRholangTemplate
+import coop.rchain.shared.Base16
+
+final case class ConfigRegistry(systemContractPubKey: String)
```
Comment (nzpr) [inline casper/src/main/scala/coop/rchain/casper/genesis/contracts/ConfigRegistry.scala]: Why this is not just `Registry`?

Comment (maximus19910112) [reply casper/src/main/scala/coop/rchain/casper/genesis/contracts/ConfigRegistry.scala]: @nzpr beacuse `Registry` module already exists in `rchain/rholang/src/main/scala/coop/rchain/rholang/interpreter/Registry/Registry.scala`

Comment (nzpr) [reply casper/src/main/scala/coop/rchain/casper/genesis/contracts/ConfigRegistry.scala]: its fine, this is the file under `casper/src/main/scala/coop/rchain/casper/genesis/contracts/` path, so it is a contract. If one wants to use both `rchain/rholang/src/main/scala/coop/rchain/rholang/interpreter/Registry/Registry.scala` and this, he can do
`import rholang.interpreter.{Registry => InterpreterRegistry}`
`import casper.genesis.contracts.{Registry => RegistryCompiledTemplate}`

At least you can use name `CompiledRegistryTemplate`, as the class extends `CompiledRholangTemplate`, but I think just `Registry` is fine

Comment (maximus19910112) [reply casper/src/main/scala/coop/rchain/casper/genesis/contracts/ConfigRegistry.scala]: @nzpr ok

**Review by zsluedem — APPROVED** [review]
LGTM

**Review by nzpr — APPROVED** [review]
LGTM

File: node/src/main/resources/defaults.conf [hunk 7]
```diff
@@ -291,6 +291,8 @@ casper {
     # Public key for transfers from the PoS vault
     # This public key is used to make transfer from PoS vault defined in PoS contract
     pos-vault-pub-key = ""
+
+    system-contract-pub-key = ""
```
Comment (tgrospic) [inline node/src/main/resources/defaults.conf:295]: @zsluedem Please create an issue to add description for this configuration parameter.

**Pull request merged**